packages/core/src/components/overlay/overlay.tsx

```diff
@@ -309,10 +309,15 @@ export class Overlay extends React.Component<IOverlayProps, IOverlayState> {
     }
 
     private handleBackdropMouseDown = (e: React.MouseEvent<HTMLDivElement>) => {
-        if (this.props.canOutsideClickClose) {
-            safeInvoke(this.props.onClose, e);
+        const { backdropProps, canOutsideClickClose, enforceFocus, onClose } = this.props;
+        if (canOutsideClickClose) {
+            safeInvoke(onClose, e);
+        }
+        if (enforceFocus) {
+            // make sure document.activeElement is updated before bringing the focus back
```

Comment:
`rAF` is decidedly _not_ "the next frame," but the last possible moment of the same frame: __before the next repaint__. it lets us execute some logic after DOM updates but before repainting (to avoid flicker).